src/components/ProgressButton.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useState } from 'react';
+import { View, StyleSheet, Text, TouchableOpacity } from 'react-native';
+import ProgressBar from 'react-native-progress/Bar';
+
+import { LinearGradient } from 'app/components';
+import { typography, palette } from 'app/styles';
+
+interface Props {
+  timeoutMilis: number;
+  stepIntervalMilis: number;
+  onComplete: () => any;
+  title: string;
+  inProgressTitle: string;
+}
+
+const noIntervalID = -1;
+
+export const ProgressButton: FC<Props> = ({
+  timeoutMilis,
```

Comment:
settingInterval on every rerender could end badly for the memory consumption and would end up probably with a memory leak. I've read the documentation and I think that setting `indeterminateAnimationDuration` to the `duration` taken from props and setting in the useEffect(() => {
 setInProgress(1)
}, [])
could work
![Zrzut ekranu 2021-01-19 o 15 32 19](https://user-images.githubusercontent.com/67273035/105048456-e5e62d80-5a6b-11eb-818e-91662eecffa2.png)

Reply:
sorry, there is no set interval on every rerender, my bad:) anyway,  please try the suggested solution if it works

Reply:
yeah, there shouldn't be any memory leaks and seems to work just fine. Refactor anyway?

Reply:
yes please, seems to be a simpler solution, no need to add custom logic if the component already has props for that. 
There could be a memory leak if the component would unmount before setInterval ends. To make it work properly there should be clearInterval on the unmounting

Reply:
when using indeterminate it looks like this (so we can't use it):
![image](https://user-images.githubusercontent.com/433674/105052314-07491880-5a70-11eb-96f4-fbf449cc70e5.png)

Reply:
ok, so please add clearInterval on the unmounting

Reply:
I've added clearInterval as a cleanup function in return from useEffect